lesson32/src/js/archive-news.js

```diff
@@ -111,13 +110,31 @@ const createThumbnail = article => {
     return thumbnailWrapper;
 };
 
-const renderArticleList = data => {
+const renderArticleList = (data, category = "all") => {
     const newsList = createElementWithClassName("ul", "news__list");
+    
+    if(category !== "all") {
+        data = filterArticlesData(data, category); 
+    }
+
     const fragment = document.createDocumentFragment();
     data.forEach(item => {
         fragment.appendChild(createArticleCards(item));
     })
-    newsContent.appendChild(newsList).appendChild(fragment);
+    newsContent.replaceChildren(newsList);
+    newsList.appendChild(fragment);
+};
+
+const filterArticlesData = (data, category) => {
+    const filteredData = data.filter(item => item.category === category);
+    return filteredData;
+};
```

Comment:
### Returns data filtered by category.

Reply:
IMO
From the function name, I thought that the variable filteredData could be omitted since the return value is easy to recognize.
```suggestion
const filterArticlesData = (data, category) => data.filter(item => item.category === category);
```

Reply:
@nakagawamai 
Thanks for the review!
I think you are right!
[5b38464](https://github.com/haru-programming/JavaScript-lessons/pull/57/commits/5b38464a8b111c6f6d7c5b08b3277eab7aaac67a)
I thought it would be simpler not to cut it out as filterArticlesData() with this modification, so I removed the function this time.